test/testsocket.py

```diff
@@ -21,7 +21,7 @@
 open_test_sockets = list()  # type: List[TestSocket]
 
 
-class TestSocket(ObjectPipe, object):
+class TestSocket(ObjectPipe[Packet], SuperSocket):
```

Comment:
I think I've tested this, but than decided to implement the Supersocket functions directly, since I had problems with the close function. Maybe that's fixed now, I just wanted to mention this, in case of strange bugs.

Reply:
This was indeed super buggy but I think I figured it out, it needed an overwrite of the `recv()` function

Reply:
Sorry for the previous ping btw 😅